mautic/mautic
# Dates added to sent emails api

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR adds dates to the API response. It also fixes the links in the corresponding dashboard widget.

Request:
```
GET /api/data/sent.email.to.contacts?dateFrom=2018-06-04&dateTo=2018-06-20&timeUnit=d&cacheTimeout=0
```

Response:
```
{
  "success": 1,
  "cached": false,
  "execution_time": 0.04371500015258789,
  "data": {
    "headItems": [
      "mautic.dashboard.label.contact.id",
      "mautic.dashboard.label.contact.email.address",
      "mautic.dashboard.label.contact.open",
      "mautic.dashboard.label.contact.click",
      "mautic.dashboard.label.contact.links.clicked",
      "mautic.dashboard.label.email.id",
      "mautic.dashboard.label.email.name",
      "mautic.dashboard.label.segment.id",
      "mautic.dashboard.label.segment.name",
      "mautic.dashboard.label.company.id",
      "mautic.dashboard.label.company.name",
      "mautic.dashboard.label.campaign.id",
      "mautic.dashboard.label.campaign.name",
      "mautic.dashboard.label.date.sent", <-- ADDED
      "mautic.dashboard.label.date.read" <-- ADDED
    ],
    "bodyItems": {
      "51": {
        "contact_id": "1",
        "contact_email": "linhartescope+contact1@gmail.com",
        "open": "1",
        "click": "1",
        "links_clicked": {
          "2018-06-06 07:47:14": "{contactfield=website}" <-- DATE KEY ADDED
        },
        "email_id": "17",
        "email_name": "token link test",
        "segment_id": "1",
        "segment_name": "linhartescope",
        "company_id": "1",
        "company_name": "AAA",
        "campaign_id": "",
        "campaign_name": "",
        "date_sent": "2018-06-06 07:45:44", <-- ADDED
        "date_read": "2018-06-06 07:47:14" <-- ADDED
      }
    }
  }
}
```

#### Steps to test this PR:
1. Execute the request with date range where you know have some email stat that has some clicks on it. Check that all dates are there.
